pkg/cli/iam_handle.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/abcxyz/access-on-demand/apis/v1alpha1"
+	"github.com/abcxyz/access-on-demand/pkg/handler"
+	"github.com/abcxyz/pkg/cli"
+	"gopkg.in/yaml.v3"
+
+	resourcemanager "cloud.google.com/go/resourcemanager/apiv3"
+)
+
+var _ cli.Command = (*IAMHandleCommand)(nil)
+
+type IAMHandleCommand struct {
+	cli.BaseCommand
+
+	flagPath string
+
+	flagDuration time.Duration
+
+	handler *handler.IAMHandler
+}
+
+func (c *IAMHandleCommand) Desc() string {
+	return `Handle the IAM request YAML file in the given path`
+}
+
+func (c *IAMHandleCommand) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+Handle the IAM request YAML file in the given path:
+
+      aod iam handle -path "/path/to/file" -duration "2h"
```

Comment:
I realized that just `duration` might be problematic. Say I changed the duration from 2h to 4h. But it shouldn't change my "start time", right? Maybe we need an optional `-start-time` that's default to `now`.

Reply:
Assume the duration change is a label change on the PR which will trigger a different workflow run, we will need to pass down the start time to the next workflow? Or the start time is just for running it locally?

Reply:
I think the start time should be the PR (aka. the request) creation time

Reply:
Taking a step back, when the iam handle workflow will be triggered? when an IAM request is approved? So I guess the start time need to be the approved time instead. When duration label is changed, will it undo the previous approval?

Reply:
Good point! Label won't undo approval

Reply:
Create an issue to add start time: https://github.com/abcxyz/access-on-demand/issues/13